master/getting-started/docker/tutorials/advanced-policy.md

```diff
@@ -2,229 +2,292 @@
+[profile]({{site.baseurl}}/{{page.version}}/reference/calicoctl/resources/profile)
+to represent each Docker network.  This profile is applied to each container 
+in that network and the profile is used by Calico to configure access policy
+for that container.  By default, the profile contains rules that allow full
+egress traffic but allow ingress traffic only from containers within the same
+network and no other source.
+
+> Custom policy for a network can be configured by creating in advance, or editing,
+> the profile associated with the Docker network.
+
+There are two ways in which the policy that defines the Docker network can be modified:
+
+-  Modify the profile policy rules.  This policy is applied directly to each container 
+   in the associated Docker network.  This approach is simple, but not very flexible, 
+   as the profile must describe the full set of rules that apply to the containers in
+   the network.
+-  Assign labels to the profile, and define global selector based policy.  The 
+   (Calico-specific) labels are assigned to containers in the associated Docker network. 
+   The globally defined policy uses selectors to determine which subset of the policy 
+   is applied to each containers based on their labels.  This approach provides a powerful
+   way to group together all of your network Policy, makes it easy to reuse policy in
+   different networks and makes it easier to define policy that extends across 
+   different orchestration systems that use Calico.
 
-When creating a Docker network using the Calico network driver, the Calico 
-driver creates a profile object for that network.  The default policy applied 
-by Calico when a new network is created allows communication between all 
-containers connected to that network, and no communication from other networks.
+## Managing Calico policy for a network
 
-Using the standard `calicoctl create` to create a `profile` resource, it is possible to 
-manage the feature-rich policy associated with a network.
+This section provides a worked examples applying policy using the two approaches
+described above.
 
-> Note that if you want to access the feature rich Calico policy, you must use
-> both the Calico Network _and_ Calico IPAM drivers together.  Using the Calico
-> IPAM driver ensures _all_ traffic from the container is routed via the host
-> vRouter and is subject to Calico policy.
+In both cases we create a Calico-Docker network and use the `calicoctl` tool to
+achieve the required isolation.
 
-> The profile that is created by the Calico network driver is given the same 
-> name as the Docker network name.
+The example we consider is this:  We want to provide isolation between a set of 
+database containers and a set of app containers, allowing only inbound
+TCP connections from the app containers to the database containers on
+port 3306.  We also want to allow all traffic between database containers.
```

Comment:
IMO there is a balance between a fully engineered solution which is gonna get quite verbose or something simple.   Certainly there should be some interconnectivity between the DB containers usually (if the DB is a cluster for example)